src/TreeAnalyzer.php

```diff
@@ -0,0 +1,185 @@
+            ) {
+                $fqn = $this->definitionResolver->resolveReferenceNodeToFqn($node);
+                if ($fqn !== null) {
+                    $this->addReference($fqn, $node);
+
+                    if (
+                        $node instanceof Node\QualifiedName
+                        && ($node->isQualifiedName() || $node->parent instanceof Node\NamespaceUseClause)
+                        && !($parent instanceof Node\Statement\NamespaceDefinition && $parent->name->getStart() === $node->getStart()
+                        )
+                    ) {
+                        // Add references for each referenced namespace
+                        $ns = $fqn;
+                        while (($pos = strrpos($ns, '\\')) !== false) {
+                            $ns = substr($ns, 0, $pos);
+                            $this->addReference($ns, $node);
+                        }
+                    }
+
+                    // Namespaced constant access and function calls also need to register a reference
+                    // to the global version because PHP falls back to global at runtime
+                    // http://php.net/manual/en/language.namespaces.fallback.php
+                    if (ParserHelpers::isConstantFetch($node) ||
+                        ($parent instanceof Node\Expression\CallExpression
+                            && !(
+                                $node instanceof Node\Expression\ScopedPropertyAccessExpression ||
+                                $node instanceof Node\Expression\MemberAccessExpression
+                            ))) {
+                        $parts = explode('\\', $fqn);
+                        if (count($parts) > 1) {
+                            $globalFqn = end($parts);
+                            $this->addReference($globalFqn, $node);
+                        }
+                    }
+                }
+            }
+        }
+        $this->collectDefinitionsAndReferences($node);
+    }
+
+    public function getDiagnostics()
+    {
+        return $this->diagnostics ?? [];
+    }
+
+    private function addReference(string $fqn, Node $node)
+    {
+        if (!isset($this->referenceNodes[$fqn])) {
+            $this->referenceNodes[$fqn] = [];
+        }
+        $this->referenceNodes[$fqn][] = $node;
+    }
+
+    public function getDefinitions()
+    {
+        return $this->definitions ?? [];
+    }
+
+    public function getDefinitionNodes()
+    {
+        return $this->definitionNodes ?? [];
+    }
+
+    public function getReferenceNodes()
+    {
+        return $this->referenceNodes ?? [];
+    }
+
+    public function getStmts()
+    {
+        return $this->stmts;
+    }
+    /**
+     * Returns the URI of the document
+     *
+     * @return string
+     */
+    public function getUri(): string
+    {
+        return $this->uri;
```

Comment:
I don't see this property and most of the above being defined or used in the class